be/src/connector/lake_connector.cpp

```diff
@@ -0,0 +1,598 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#include "connector/lake_connector.h"
+
+#include "common/constexpr.h"
+#include "exec/vectorized/connector_scan_node.h"
+#include "exec/vectorized/olap_scan_prepare.h"
+#include "gen_cpp/InternalService_types.h"
+#include "runtime/global_dict/parser.h"
+#include "storage/column_predicate_rewriter.h"
+#include "storage/conjunctive_predicates.h"
+#include "storage/lake/tablet_manager.h"
+#include "storage/lake/tablet_reader.h"
+#include "storage/predicate_parser.h"
+#include "storage/projection_iterator.h"
+#include "util/starrocks_metrics.h"
+
+namespace starrocks::connector {
+using namespace vectorized;
+
+// ================================
+
+class LakeDataSourceProvider;
+
+// TODO: support parallel scan within a single tablet
+class LakeDataSource final : public DataSource {
+public:
+    explicit LakeDataSource(const LakeDataSourceProvider* provider, const TScanRange& scan_range);
+    ~LakeDataSource() override;
+
+    Status open(RuntimeState* state) override;
+    void close(RuntimeState* state) override;
+    Status get_next(RuntimeState* state, vectorized::ChunkPtr* chunk) override;
+
+    int64_t raw_rows_read() const override { return _raw_rows_read; }
+    int64_t num_rows_read() const override { return _num_rows_read; }
+    int64_t num_bytes_read() const override { return _bytes_read; }
+
+    // TODO:
+    // Return last bytes of Scan or Exchange Data, then reset it
+    // int64_t last_spent_cpu_time_ns() override;
```

Comment:
Please refer this PR to implement the `get_cpu_time` interface: https://github.com/StarRocks/starrocks/pull/8088